### ISSUE TITLE
Analytics is crashing Unity on mac

### ISSUE THREAD

**Issue by aholla** [firebase/quickstart-unity#943]
Hi, im using Firebase 7.0.2 with unity 2020.1.17f.

After i have initialised firebase, any calls to "FirebaseAnalytics" cause Unity to hang with a spinny wheel.

Do you have any ideas what might be wrong?

I have Core 7.0.2, Analytics 7.0.2 and Crashlytics 7.0.2

Here is some sample code:

``` 
private static void Init()
{
    FirebaseApp.CheckAndFixDependenciesAsync().ContinueWithOnMainThread(task =>
    {
        m_dependencyStatus = task.Result;

        if (m_dependencyStatus == DependencyStatus.Available)
        {
            InitializeFirebase();
        }
        else
        {
            Debug.LogError($"Could not resolve all Firebase dependencies: {m_dependencyStatus}");
        }
    });
}
```
```
private static void InitializeFirebase()
{
    // This method or the one below will cause Unity to hang.
    FirebaseAnalytics.SetAnalyticsCollectionEnabled(true);

    // Set default session duration values.
    FirebaseAnalytics.SetSessionTimeoutDuration(new TimeSpan(0, 30, 0));
}
```

**Comment by DellaBitta** [comment]
Hi @aholla,

I attempted to reproduce this with our Analytics quickstart app in this repo and I didn't receive a crash or hang.  I used the Unity version and Firebase SDK version that you reported above.  Can you attempt to reproduce this with the quickstart and it's initialization flow?

Thanks.

**Comment by aholla** [comment]
I've tried loading a new project but I'm getting a security error when I try and run the project. 

'"FirebaseCppApp-7_0_2.bundle" cannot be opened because the developer cannot be verified....'

This pops up several times and then goes away but leaves the following:

'DllNotFoundException: FirebaseCppApp-7_0_2'

I've not had this before. I downloaded the SDK but in my other project, I have the sdk via the package manager. Im not sure how that happened.

**Comment by DellaBitta** [comment]
Hi @aholla,

I believe I know that's going wrong.  Recent versions of MacOS embed a security feature which detect binaries that weren't distributed from the Apple Store.  This includes the Firebase SDK which was downloaded from our website instead of the store itself.  When prompted with the dialog box you've encountered you must go through some steps in the system settings to enable the OS to load the binary and execute it properly.  Full details can be [found here](https://howtoremove.guide/macos-cannot-verify/), but I've digested the major points below and edited them for context:

```
1: Either go to `System Preferences` and open `Security & Privacy`, or, when the dialog box appears warning of loading FirebaseCppApp, click on the `?` in the lower left hand corner.
1: In the `General Tab`, under `Allow apps downloaded from`, you should see the FirebaseCPPApp at the bottom.  Click on  `Open Anyway`.  You will probably need to unlock this option by clicking on the lock in the lower left hand corner and entering your password.
1: You will once again see a warning message that tells you the app cannot be verified as safe, but this time you will have the option to open it anyway. 
```

You might need to execute step 2, above, a few times, depending on which libraries are being loaded, and/or restart the unity app that's attempting to run on your mac after you've accepted all of the dialog popups.

I hope this helps.

**Comment by Smoothjazzy** [comment]
I'm having a similar problem to @aholla. It's almost exactly the same problem as here https://github.com/firebase/quickstart-unity/issues/807
where Unity crashes on second play after updating Firebase. Adjusting the Mac Security and Privacy settings has not fixed the issue. 
I'm using Firebase 7.0.2 (Analytics and Auth) and Unity 2019.4.18

**Comment by aholla** [comment]
Sorry, i didnt have enough time to stick around and so went with another sdk.

**Comment by chkuang-g** [comment]
@aholla Sorry to hear that. If you have a chance to try Firebase SDK again, and still encounter the issue, please open a new ticket.  I will close this one for now.

@Smoothjazzy In short, I would recommend you to open a new ticket and fill in your information from the template, like your Firebase version and Unity version, etc.
#807 should have been fixed in `6.16.1`. Also, that is related to Database and Firestore, not Analytics.  It does seems like your issue many not be the same to #807.
And again, please open a new ticket.  It is really easier for us to track issue this way.

Shawn
